iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java

```diff
@@ -2478,7 +2478,19 @@ public int getDefaultStorageGroupLevel() {
     return defaultStorageGroupLevel;
   }
 
-  void setDefaultStorageGroupLevel(int defaultStorageGroupLevel) {
+  void setDefaultStorageGroupLevel(int defaultStorageGroupLevel, boolean startUp) {
+    if (defaultStorageGroupLevel < 1) {
+      if (startUp) {
+        logger.warn(
+            "Illegal defaultStorageGroupLevel: {}, should >= 1, use default value 1",
+            defaultStorageGroupLevel);
+        defaultStorageGroupLevel = 1;
+      } else {
+        throw new IllegalArgumentException(
```

Comment:
If the configuration file contains an value less than 1, it is allowed at startup, but it may cause failure during hot loading. Why not handle it the same way during hot loading as startup?

Reply:
First, start-up failures should be avoided as much as possible because users tend not to check the logs after start-up failures and directly throw the problems to us.
However, if we simply ignore the illegal value during hot-load, the user may think the statement is successful, and the configuration has been changed, which is also unexpected.